src/server/request_handler/default_request_handler.ts

```diff
@@ -370,6 +373,12 @@ export class DefaultRequestHandler implements A2ARequestHandler {
         }
 
         const latestTask = await this.taskStore.load(params.id);
+        if (!latestTask) {
+            throw A2AError.internalError(`Task ${params.id} not found after cancellation.`);
+        }
+        if (latestTask.status.state != "canceled") {
+            throw A2AError.taskNotCancelable(params.id);
```

Comment:
Can we include resulting state into the error message as [in python](https://github.com/a2aproject/a2a-python/blob/aa159f3e1076ae6eaad5576119d7857c2a9b2448/src/a2a/server/request_handlers/default_request_handler.py#L181)? I think it's useful for debugging.